dashboard/lti_new.py

```diff
@@ -64,17 +69,40 @@ def check_if_success_getting_tool_config(tool_config):
         logger.info('Success in fetching LTI tool configuration')
         return True
     else:
-        logger.error(tool_config)
+        logger.error(f'Invalid LTI configuration: "{tool_config}"')
         return False
 
 
-def get_jwks(request):
+def get_jwks(_):
     config = get_tool_conf()
     if not check_if_success_getting_tool_config(config):
         return LTIError(config).response_json()
     return JsonResponse(config.get_jwks())
 
 
+def generate_config_json(request: HttpRequest) -> \
+        Union[HttpResponse, LTIError]:
+    config = get_tool_conf()
+    if not check_if_success_getting_tool_config(config):
+        return LTIError(f'Invalid LTI configuration: "{config}"')\
+            .response_json()
+
+    parameters = {
+        'timestamp': datetime.now().isoformat(),
+        'host': request.get_host(),
+        'base_url': urllib.parse.urlunsplit(
+            [request.scheme, request.get_host()] + 3 * ['']),
+        'login_url_suffix': reverse('login'),
+        'launch_url_suffix': reverse('launch'),
+        'jwks_url_suffix': reverse('get_jwks'),
+    }
+
+    return HttpResponse(
+        open(finders.find(
+            'config/lti_config_template.json')).read() % parameters,
```

Comment:
I though we want to upload the `lti_config_template.json` as secret  as in `env.json`. so that we can change it on fly if we have to. Because all the file that are place in `config` are copied to a local `mylasecrets` and in docker-compose `${HOME}/mylasecrets:/secrets`. 
@jonespm Thoughts???

Reply:
@pushyamig, if that's what you meant, I didn't understand your comments.

In my comment at (https://github.com/tl-its-umich-edu/my-learning-analytics/pull/1105#discussion_r505579885), I asked for clarification.  You chose the "Move LTI config template to assets/config/lti_config_template.json" option and gave the comment a "thumbs up" reaction.

I also repeated the option in the comment at (https://github.com/tl-its-umich-edu/my-learning-analytics/pull/1105#discussion_r505583005) so there would not be any misunderstanding.

Putting the template into a secret is overkill.  It doesn't contain anything that would be considered "secret" and it's unlikely to be changed.

Reply:
@lsloan Sorry for confusing comment. But Please keep the file as in the config directory and it's fine. I do agree we don't change the template file that often and if we do have to change something in emergency we can do it in canvas directly. This template is only used during initial creating time of LTI Key.